presto-spark-base/src/main/java/com/facebook/presto/spark/planner/PrestoSparkRddFactory.java

```diff
@@ -206,6 +208,17 @@ private PlanFragment configureOutputPartitioning(Session session, PlanFragment f
         return fragment;
     }
 
+    private static JavaPairRDD<MutablePartitionId, PrestoSparkMutableRow> partitionBy(JavaPairRDD<MutablePartitionId, PrestoSparkMutableRow> rdd, Partitioner partitioner)
+    {
+        JavaPairRDD<MutablePartitionId, PrestoSparkMutableRow> javaPairRdd = rdd.partitionBy(partitioner);
+        ShuffledRDD<MutablePartitionId, PrestoSparkMutableRow, PrestoSparkMutableRow> shuffledRdd = (ShuffledRDD<MutablePartitionId, PrestoSparkMutableRow, PrestoSparkMutableRow>) javaPairRdd.rdd();
```

Comment:
Now I understand why @sameeragarwal  once recommended to use Scala. Spark JavaRDD API is much more restricted so we often have to "unwrap" `JavaRDD` , do something on Scala RDD, and then "re-wrap" into a Java RDD...

Reply:
Yeah. I also notices that transforming Scala iterators to Java iterators and back is inefficient, as every transform produces one more frame on the stack (virtual functions cannot be inlined). I'm going to address the iterators problem in the next PR.